hpctestlib/apps/cp2k/__init__.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2016-2021 Swiss National Supercomputing Centre (CSCS/ETH Zurich)
+# ReFrame Project Developers. See the top-level LICENSE file for details.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+import reframe as rfm
+import reframe.utility.sanity as sn
+import reframe.utility.typecheck as typ
+
+__all__ = ["Cp2k"]
+
+
+class Cp2k(rfm.RunOnlyRegressionTest):
+    '''Base class for the CP2K Test.
+
+    CP2K is a quantum chemistry and solid state physics software
+    package that can perform atomistic simulations of solid state,
+    liquid, molecular, periodic, material, crystal, and biological
+    systems. CP2K provides a general framework for different modeling
+    methods such as DFT using the mixed Gaussian and plane waves
+    approaches GPW and GAPW.  (see cp2k.org).
+
+    The presented abstract run-only class checks the work of cp2k.
+    To do this, it is necessary to define in tests  the reference
+    values of energy and possible deviations from this value.
+    This data is used to check if the task is being executed
+    correctly, that is, the final energy is correct
+    (approximately the reference). The default assumption
+    is that CP2K is already installed on the device under test.
+    '''
+
+    #: Reference value of energy, that is used for the comparison
+    #: with the execution ouput on the sanity step. Final value of
+    #: energy should be approximately the same
+    #:
+    #: :default: :class:`required`
+    energy_value = variable(float)
```

Comment:
For consistency with amber:
```suggestion
    energy_reference = variable(float)
```

Reply:
For cp2k, we provide a single case, so we can set the entry references and tolerances directly here in the base class.